### PR TITLE
Don't show ips to non-administrators, Fixes #694

### LINKED ISSUE #694: Optional scrubbing of ip addresses from rcon console
Since ip addresses are considered sensitive / confidential information it's reasonable to hide them from non-administrators (moderators in our case).

My idea is to implement a feature that optionally allows the hiding of ip addresses from specific 'rcon authorities' (like moderators).

### PR DESCRIPTION
**Pull request by Ryozuki**
People with non-administrator rights won't be able to see ips (mod, helper).

### CODE REVIEW

**Comment by Learath2** [comment]
There is probably something related to mute you are missing. Otherwise looks good.

**Comment by heinrich5991** [comment]
There's also a bit missing wrt. players joining and leaving. If you want to solve it this way, you could search for `net_addr_str` (and `GetClientAddr`) in the code base and review each use.

**Comment by Ryozuki** [comment]
Now it should cover all cases, also i renamed the var c to a more descriptive name "ClientID" :)

**Comment by Learath2** [comment]
Well I might've used a wrapper for `net_addr_str` which turns it into `xxx.xxx.xxx.xxx` but this works too.

**Comment by heinrich5991** [comment]
The newly added cases (player joining, etc.) only show the IP addresses of admins to everyone, instead of only showing the IP addresses of everyone to admins.

File: src/engine/server/server.cpp [hunk 1]
```diff
@@ -841,8 +841,8 @@ int CServer::DelClientCallback(int ClientID, const char *pReason, void *pUser)
 
 	char aAddrStr[NETADDR_MAXSTRSIZE];
 	net_addr_str(pThis->m_NetServer.ClientAddr(ClientID), aAddrStr, sizeof(aAddrStr), true);
-	char aBuf[256];
-	str_format(aBuf, sizeof(aBuf), "client dropped. cid=%d addr=%s reason='%s'", ClientID, aAddrStr,	pReason);
+
+	char aBuf[256];str_format(aBuf, sizeof(aBuf), "client dropped. cid=%d addr=%s reason='%s'", ClientID, aAddrStr, pReason);
```
Comment (heinrich5991) [inline src/engine/server/server.cpp]: A newline was accidently deleted here.

File: src/engine/server/server.cpp [hunk 2]
```diff
@@ -1802,44 +1802,51 @@ int CServer::Run()
...
+							bool CanSeeAddress = m_aClients[ClientID].m_Authed > CServer::AUTHED_MOD;
 
 							char aBuf[256];
-							str_format(aBuf, sizeof(aBuf), "ClientID=%d addr=%s secure=%s blacklisted", c, aAddrStr, m_NetServer.HasSecurityToken(c)?"yes":"no");
+
+							if(CanSeeAddress)
+								str_format(aBuf, sizeof(aBuf), "ClientID=%d addr=%s secure=%s blacklisted", ClientID, aAddrStr, m_NetServer.HasSecurityToken(ClientID)?"yes":"no");
+							else
+								str_format(aBuf, sizeof(aBuf), "ClientID=%d secure=%s blacklisted", ClientID, m_NetServer.HasSecurityToken(ClientID) ? "yes" : "no");
+
```
Comment (heinrich5991) [inline src/engine/server/server.cpp]: This still only shows the IP addresses of admins, and of no one else, if I read the code correctly.

**Comment by Ryozuki** [comment]
I finally got it, wathever string you print to the level IConsole::OUTPUT_LEVEL_STANDARD will be shown to all staff levels,

But if you print to the IConsole::OUTPUT_LEVEL_ADDINFO only rcon admins and server console will see it

**Comment by heinrich5991** [comment]
> I finally got it, wathever string you print to the level IConsole::OUTPUT_LEVEL_STANDARD will be shown to all staff levels,

What I read from the code was that no one sees console output unless you're admin *or* someone has just executed a command on your level.

**Comment by heinrich5991** [comment]
Thanks!

**Pull request merged**